src/Microsoft.AspNetCore.Blazor.Build/ReferenceFromSource.props

```diff
@@ -29,6 +29,11 @@
     <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="$(RazorPackageVersion)"/>
   </ItemGroup>
 
+  <!-- This is used as a P2P when building the repo. Normal Blazor projects will get this as a reference through the Blazor.Build package -->
+  <ItemGroup>
+    <ProjectReference Include="$(MSBuildThisFileDirectory)..\Microsoft.AspNetCore.Blazor.TagHelperWorkaround\Microsoft.AspNetCore.Blazor.TagHelperWorkaround.csproj" />
+  </ItemGroup>
```

Comment:
I'm guessing then that `ReferenceFromSource.props` inherently behaves like "privateassets=all", otherwise this change would break the "HostedInAspNet" sample (because the server project references the client, which imports `ReferenceFromSource.props` and therefore obtains a reference to TagHelperWorkaround).

Please let me know if I'm misunderstanding this!

Reply:
So it turns out that p2p references support privateassets as well, which is needed here.